src/states/node.rs

```diff
@@ -1654,6 +1736,64 @@ impl Node {
         true
     }
 
+    fn send_own_group_merge(&mut self,
+                            targets: Vec<XorName>,
+                            merge_details: OwnMergeDetails<XorName>,
+                            src: Authority) {
+        for target in &targets {
+            let (connected, needed): (Vec<_>, Vec<_>) = merge_details.groups
+                .clone()
+                .into_iter()
+                .partition(|&(prefix, _)| merge_details.prefix.is_compatible(&prefix));
```

Comment:
I don't understand this partition: The recipient is not necessarily already connected to everyone in a group compatible with `merge_details.prefix`: E.g. if `00` is merging with the four `01??` groups, `0100` and `0111` won't be connected yet.

On the other hand, the recipient may well already be connected to groups that are not selected here: E.g. `0100` will be connected to `1`, if it exists.

But wouldn't it be easier anyway to only make that distinction on the side of the recipient? The sender could just send every new group member the same information, and they could check for themselves whom they are already connected to?

Also, I think the recipient will need to send another `OwnGroupMerge` to everyone else in the group, because they might know a neighbouring group that the other constituent groups don't know yet. (Or do we currently rely on the `OtherGroupMerge` handling for that?)